multidict/_multidict_c.c

```diff
@@ -0,0 +1,854 @@
+#include <Python.h>
+#include <structmember.h>
+
+#include "_pair_list.h"
+#include "_multidict_views.h"
+
+// fix for VisualC complier used by Python 3.4
+#ifdef __GNUC__
+#define INLINE inline
+#else
+#define INLINE
+#endif
+
+static PyObject *collections_abc_mapping;
+static PyObject *istr;
+
+static PyTypeObject multidict_type;
+static PyTypeObject multidict_proxy_type;
+
+#define MultiDict_Check(o) (Py_TYPE(o) == &multidict_type)
+
+typedef struct {
+    PyObject_HEAD
+    PyObject *impl;
+} _MultiDictObject;
+
+/******************** Base Methods ********************/
+
+/* Forward declaration */
+static PyObject *multidict_items(_MultiDictObject *self);
+
+static INLINE PyObject *
+_multidict_getone(_MultiDictObject *self, PyObject *key, PyObject *_default)
+{
+    PyObject *val = pair_list_get_one(self->impl, key);
```

Comment:
Ideally, we should avoid setting an exception if not found and clearing the exception on the next step.